Sources/SwiftToolchain/Toolchain.swift

```diff
@@ -283,21 +283,21 @@ public final class Toolchain {
     return testBundlePath
   }
 
-  public func packageInit(name: String, type: PackageType, inPlace: Bool) throws {
-    var initArgs = [
+  public func runPackageInit(name: String, type: PackageType, initialiseInSubdirectory: Bool) throws {
+    var arguments = [
       swiftPath.pathString, "package", "init",
       "--type", type.rawValue,
     ]
-    if !inPlace {
-      initArgs.append(contentsOf: ["--name", name])
-    }
-    try ProcessRunner(initArgs, terminal)
-      .waitUntilFinished()
+	
+	if initialiseInSubdirectory {
+		arguments += ["--name", name]
+	}
+	
+    try ProcessRunner(arguments, terminal).waitUntilFinished()
   }
```

Comment:
This function can be made simpler if we replace it with something like this:

```swift
public func initialisePackage(type: PackageType, for project: Project) throws {
    try localFileSystem.createDirectory(project.path) // should already be created in `Init.run()`
    guard let currentDirectory = localFileSystem.currentWorkingDirectory else {
        throw someError
    }
    let projectRelativePath = project.path.relative(to: currentDirectory)
    let arguments = [
        swiftPath.pathString, "package", "init",
        "--type", type.rawValue,
        "--name", projectRelativePath.pathString
    ]
    try ProcessRunner(arguments, terminal).waitUntilFinished()
}
```

If this function is simplified, other functions that call it can be simplified as well, without having to carry the extra `initialiseInSubdirectory` information. However, I'm not sure if this is making too much assumption around the current directory path and the project's path.

Reply:
Yeah, what if `projectRelativePath` evaluates to `.`?

Reply:
I experimented a bit. `.` and `..` and anything but a single undelimited path component is not going to work. 

If I do `swift package init --name a/b/../c`, then it will create a structure like this:

```
.
├── Package.swift
├── README.md
├── Sources
│   └── a
│       ├── b
│       └── c
│           └── main.swift
└── Tests
    ├── LinuxMain.swift
    └── a
        └── cTests
            ├── XCTestManifests.swift
            └── a_b____cTests.swift
```

I also found that SwiftPM always creates new projects "in place" and never creates a project in a subdirectory of the current directory. Should carton mirror what SwiftPM does?

Reply:
I hope that by delegating to `swift package init` as much as possible we would essentially mirror their behavior, right?

Reply:
Yep, I think so. Although, by mirroring SwiftPM, carton will have to remove the ability of creating new project in a subdirectory. I don't know if this is a right thing to do.

If carton should behave like SwiftPM, then we can probably just special-case `.` for `projectRelativePath`.

Reply:
Sorry, I'm a bit confused. Wouldn't the simplified version, as you've shown in `initialisePackage` in the original comment here, mirror SwiftPM behavior completely?

Reply:
I think I might be confusing myself here as well. 

My understanding is that there is a difference between these 2 commands: 

`swift package init`

`swift package init --name .`

The first command initialises a project with using the name of the currently working directory. Whereas the second command initialises a project named "." and with "." replaced by "_" in source file names.

In carton's logic (both in trunk and in this PR), when the user sets the `--name` option in `carton init`, the project's path is recorded as `currently/working/directory/projectName`. So if the user does `carton init --name .`, then the project's path evaluates to the current directory.

As I'm writing this comment, I realise that `initialisePackage` can completely mirror SwiftPM's behaviour, if I change `Project` to include only the project's name, but not its path. And there is no need for special-casing `.`. Currently in this PR, `Project` includes only the path, and `initialisePackage` is not able to distinguish between `carton init` and `carton init --name .` from just the path.

Reply:
Right, I think that `--name .` is such a rare case that we're unlikely to see anyone stumbling upon this inconsistency in practice. But I also don't mind if you clean it up in this PR or separately, or not at all given that it's an obscure edge case.